Tribler/Test/test_as_server.py

```diff
@@ -125,6 +125,11 @@ def tearDown(self, annotate=True):
         self.assertFalse(delayed_calls, "The reactor was dirty when tearing down the test")
         self.assertFalse(Session.has_instance(), 'A session instance is still present when tearing down the test')
 
+        # Clean the delayed calls to prevent them cascading into the next test case
+        for delayed_call in reactor.getDelayedCalls():
```

Comment:
I see you haven't even tested it. there's an assertion at line 125 that makes it impossible to reach your code. 

Just add the cancel call after line 124.

Please, test everything you send.